src/Services/EncounterService.php

```diff
@@ -242,6 +256,69 @@ public function getEncounterForPatient($pid, $eid)
         return sqlQuery($sql, array($pid, $eid));
     }
 
+    public function insertEncounter($pid, $data)
+    {
+
+        $encounter = generate_id();
+        
+    
+        $sql .= " INSERT INTO form_encounter SET";
+        $sql .= "     date = ?,";
+        $sql .= "     onset_date = ?,";
+        $sql .= "     reason = ?,";
+        $sql .= "     facility = ?,";
+        $sql .= "     pc_catid = ?,";
+        $sql .= "     facility_id = ?,";
+        $sql .= "     billing_facility = ?,";
+        $sql .= "     sensitivity = ?,";
+        $sql .= "     referral_source = ?,";
+        $sql .= "     pid = ?,";
+        $sql .= "     encounter = ?,";
+        $sql .= "     pos_code = ?,";
+        $sql .= "     external_id = ?,";
+        $sql .= "     provider_id = ?,";
+        $sql .= "     parent_encounter_id = ?";
+        
+      $results;
+        addForm(
+            $encounter,
+            "New Patient Encounter",
+           $results = sqlInsert(
+            $sql,
+            [
+                $data["date"],
+                $data["onset_date"],
+                $data["reason"],
+                $data["facility"],
+                $data["pc_catid"],
+                $data["facility_id"],
+                $data["billing_facility"],
+                $data["sensitivity"],
+                $data["referral_source"],
+                $pid,
+                $encounter,
+                $data["pos_code"],
+                $data["external_id"],
+                $data["provider_id"],
+                $data["parent_enc_id"]
+            ]
+            ),
+            "newpatient",
+            $pid,
+            $data["provider_id"],
+            $data["date"]
+        );
+
+        if ($results) {
+            return $results;
+        }
+
```

Comment:
reduce these empty lines to one empty line

Reply:
Done